flink-libraries/flink-ml/src/main/scala/org/apache/flink/ml/recommendation/ALS.scala

```diff
@@ -122,12 +121,26 @@ import scala.util.Random
   * [[https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/mllib/
   * recommendation/ALS.scala here]].
   */
-class ALS extends Predictor[ALS] {
+class ALS extends Predictor[ALS] with RankingPredictor[ALS] with TrainingRatingsProvider {
```

Comment:
`ALS` is also a `RankingPredictor` now, and also `TrainingRatingsProvider` for excluding known items from rankings.